### PR TITLE
Bugfix: sync with extension

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

This PR tries to fix a bug on latest public beta release where users are seeing an outdated qr code alert
- Updated timeout to 2000 in case of slower communication
- Better error logs for this particular case
- Now if first attempt to sync after establishing connection fails, we try it again

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/Views/ImportWallet/index.js [hunk 1]
```diff
@@ -153,6 +153,23 @@ class ImportWallet extends PureComponent {
...
+			await this.pubnubWrapper.startSync();
+			return true;
+		} catch (e) {
+			!firstAttempt && this.props.navigation.goBack();
+			!firstAttempt &&
+				Alert.alert(
+					strings('sync_with_extension.outdated_qr_code'),
+					strings('sync_with_extension.outdated_qr_code_desc')
+				);
+			Logger.error('Sync::startSync', firstAttempt);
```
Comment (brunobar79) [inline app/components/Views/ImportWallet/index.js]: the second parameter needs to be the error, not the attempt. 
Should be something like this:
```
Logger.log('Sync::startSync', firstAttempt);
Logger.log('SyncError::startSync', e.toString());
Logger.error('SyncError::startSync', e);
```

File: app/components/Views/SyncWithExtension/index.js [hunk 2]
```diff
@@ -143,6 +143,23 @@ class SyncWithExtension extends PureComponent {
...
+			await this.pubnubWrapper.startSync();
+			return true;
+		} catch (e) {
+			!firstAttempt && this.props.navigation.goBack();
+			!firstAttempt &&
+				Alert.alert(
+					strings('sync_with_extension.outdated_qr_code'),
+					strings('sync_with_extension.outdated_qr_code_desc')
+				);
+			Logger.error('Sync::startSync', firstAttempt);
```
Comment (brunobar79) [inline app/components/Views/SyncWithExtension/index.js]: Same

File: app/util/syncWithExtension.js [hunk 3]
```diff
@@ -139,13 +140,14 @@ export default class PubNubWrapper {
 		this.pubnub.addListener({
 			message: ({ channel, message }) => {
 				if (channel !== this.channelName || !message) {
+					Logger.error('Sync::message', channel !== this.channelName, !message);
```
Comment (brunobar79) [inline app/util/syncWithExtension.js]: Logger.log first with the info.
Logger.error later with an error or object

File: app/util/syncWithExtension.js [hunk 4]
```diff
@@ -139,13 +140,14 @@ export default class PubNubWrapper {
...
 					this.timeout = false;
 					return false;
 				}
 				if (message.event === 'error-sync') {
 					this.timeout = false;
 					this.disconnectWebsockets();
-					Logger.error('Sync failed', message, this.incomingDataStr);
+					Logger.error('Sync::error-sync', message, this.incomingDataStr);
```
Comment (brunobar79) [inline app/util/syncWithExtension.js]: same

Comment (brunobar79) [reply app/util/syncWithExtension.js]: There might be sensitiveInfo on this.incomingString, you don't want to log that

File: app/util/syncWithExtension.js [hunk 5]
```diff
@@ -156,7 +158,7 @@ export default class PubNubWrapper {
 							const data = JSON.parse(this.incomingDataStr);
 							onSyncingData(data);
 						} catch (e) {
-							Logger.error('Sync failed at parsing', e);
```
Comment (brunobar79) [inline app/util/syncWithExtension.js]: Here you might wanna log the message object first, then the error

**Review by brunobar79 — APPROVED** [review]
Some comments about logging but 👍

File: app/util/syncWithExtension.js (hunk 5 above)
Comment (brunobar79) [reply app/util/syncWithExtension.js]: But first delete the data which is not relevant in this case.

**Review by ibrahimtaveras00 — APPROVED** [review]
looks good on both android and iOS, QA Passed 👍

File: app/util/syncWithExtension.js [hunk 6]
```diff
@@ -139,13 +140,15 @@ export default class PubNubWrapper {
...
+					Logger.log('Sync::message', channel !== this.channelName, !message);
+					Logger.error('Sync::message', channel !== this.channelName, !message);
 					this.timeout = false;
 					return false;
 				}
 				if (message.event === 'error-sync') {
 					this.timeout = false;
 					this.disconnectWebsockets();
-					Logger.error('Sync failed', message, this.incomingDataStr);
+					Logger.error('Sync::error-sync');
```
Comment (brunobar79) [inline app/util/syncWithExtension.js]: Not 100% sure but I think the second parameter (object | error ) is required

Comment (estebanmino) [reply app/util/syncWithExtension.js]: `incomingDataStr` is the data parsed three lines below if the sync succeeds, so is either all info from the user or an empty string

**Pull request merged**